### PR TITLE
[Tracing] Change import path of trace

### PR DESCRIPTION
**Pull request by lumoslnt**
# Description

In this PR, we change the import path of trace to `promtpflow.tracing`, in order to prepare for splitting the promptflow-tracing package from promptflow.

# All Promptflow Contribution checklist:
- [x] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow/promptflow/tracing/tracer.py [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+from promptflow._core.generator_proxy import GeneratorProxy, generate_from_proxy
+from promptflow._core.thread_local_singleton import ThreadLocalSingleton
+from promptflow._utils.dataclass_serializer import serialize
+from promptflow.contracts.tool import ConnectionType
+
+from .._utils.utils import default_json_encoder
+from .contracts.trace import Trace, TraceType
+
+
+class Tracer(ThreadLocalSingleton):
```
Comment (wangchao1230) [inline src/promptflow/promptflow/tracing/tracer.py:19]: is this mean to public exposed? change file to _tracer if not.

File: src/promptflow/promptflow/_core/openai_injector.py [hunk 2]
```diff
@@ -13,9 +13,9 @@
 import openai
 
 from promptflow._core.operation_context import OperationContext
-from promptflow.contracts.trace import TraceType
+from promptflow.tracing._trace import _traced_async, _traced_sync
```
Comment (wangchao1230) [inline src/promptflow/promptflow/_core/openai_injector.py:16]: we should also move this file to promptflow.tracing?

Comment (thy09) [reply src/promptflow/promptflow/_core/openai_injector.py:16]: Let's do it in another PR

**Review by thy09 — DISMISSED** [review]

File: src/promptflow/promptflow/tracing/tracer.py (hunk 1 above)
Comment (lumoslnt) [reply src/promptflow/promptflow/tracing/tracer.py]: Changed to `_tracer`.

File: src/promptflow/tests/executor/unittests/_core/test_api_injector.py [hunk 3]
```diff
@@ -21,13 +21,13 @@
...
 )
 from promptflow._core.operation_context import OperationContext
-from promptflow._core.tracer import Tracer
 from promptflow._version import VERSION
 from promptflow.connections import AzureOpenAIConnection
-from promptflow.contracts.trace import TraceType
 from promptflow.exceptions import UserErrorException
 from promptflow.tools.aoai import AzureOpenAI
 from promptflow.tools.embedding import embedding
+from promptflow.tracing.contracts.trace import TraceType
```
[the remark this answers is not in the document]
Comment (lumoslnt) [reply src/promptflow/tests/executor/unittests/_core/test_api_injector.py:30]: Will do in another PR.

**Review by thy09 — APPROVED** [review]

**Review by wangchao1230 — APPROVED** [review]

**Pull request merged**